build-android.sh

```diff
@@ -1,159 +1,92 @@
 #!/bin/bash
 
-# This script builds the iOS and Mac OpenSSL libraries with Bitcode enabled
-
-# Credits:
-# https://github.com/sqlcipher/android-database-sqlcipher/blob/gradle/android-database-sqlcipher/build-openssl-libraries.sh
-
-MINIMUM_ANDROID_SDK_VERSION=21
-MINIMUM_ANDROID_64_BIT_SDK_VERSION=21
-
-(cd openssl;
-
-if [ ! ${ANDROID_NDK_ROOT} ]; then
-    echo "ANDROID_NDK_ROOT environment variable not set, set and rerun"
+if [ ! ${ANDROID_NDK_HOME} ]; then
+    echo "ANDROID_NDK_HOME environment variable not set, set and rerun"
     exit 1
 fi
 
-echo "----------------------------------------"
-echo "Building OpenSSL for Android"
-echo " "
-echo "Android SDK version: ${MINIMUM_ANDROID_SDK_VERSION}"
-echo "Android 64 bit SDK Version: ${MINIMUM_ANDROID_64_BIT_SDK_VERSION}"
-echo "Android NDK Root: ${ANDROID_NDK_ROOT}"
-echo "----------------------------------------"
-echo " "
+# Set directory
+SCRIPTPATH=`realpath .`
+OPENSSL_DIR=$SCRIPTPATH/openssl
 
-ANDROID_LIB_ROOT=../lib/android
-TMP_DIR=/tmp/openssl/android
-ANDROID_TOOLCHAIN_DIR=${TMP_DIR}/toolchain
-OPENSSL_CONFIGURE_OPTIONS="no-krb5 no-idea no-camellia \
-        no-seed no-bf no-cast no-rc2 no-rc4 no-rc5 no-md2 \
-        no-md4 no-ripemd no-rsa no-ecdh no-sock no-ssl2 no-ssl3 \
-        no-dsa no-dh no-ec no-ecdsa no-tls1 no-pbe no-pkcs \
-        no-tlsext no-pem no-rfc3779 no-whirlpool no-ui no-srp \
-        no-ssltrace no-tlsext no-mdc2 no-ecdh no-engine \
-        no-tls2 no-srtp -fPIC"
+cd "${OPENSSL_DIR}" || exit 1
 
+# Find the toolchain for your build machine
 HOST_INFO=`uname -a`
 case ${HOST_INFO} in
     Darwin*)
-        TOOLCHAIN_SYSTEM=darwin-x86
+        toolchains_path=${ANDROID_NDK_HOME}/toolchains/llvm/prebuilt/darwin-x86_64
         ;;
     Linux*)
-        if [[ "${HOST_INFO}" == *i686* ]]
-        then
-            TOOLCHAIN_SYSTEM=linux-x86
-        else
-            TOOLCHAIN_SYSTEM=linux-x86_64
-        fi
+        toolchains_path=${ANDROID_NDK_HOME}/toolchains/llvm/prebuilt/linux-x86_64
         ;;
     *)
         echo "Toolchain unknown for host system"
         exit 1
         ;;
 esac
```

Comment:
I know this is just tweaking what was there before, but I'm guessing this won't work on the M1 mac. Happy to test that out, but I think with some alternate uses of `uname` we can simplify this a bit

```suggestion
toolchains_path=/toolchains/llvm/prebuilt/$(uname -s | tr A-Z a-z)-$(uname -m)
```

Or maybe 

```suggestion
OS_TYPE=$(uname -s | tr A-Z a-z)
ARCH=$(uname -m)
toolchains_path=/toolchains/llvm/prebuilt/$OS_TYPE-$ARCH
```

Reply:
Sorry, sugggestion is slightly wrong, forgot the `${ANDROID_NDK_HOME}` from the `toolchains_path`. Got too excited with the fancy tools 😄